### ISSUE TITLE
Error on Windows during self update in cmd.exe

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang-nursery/multirust-rs#230]
This morning I ran `rustup` in a new cmd.exe window. Everything succeeded until the very end where it attempted a self update. The main process exited, I got my shell prompt back, and then the following message was printed:

```
failure during windows uninstall: The handle is invalid (os error 6)
```

This was reproducible in `cmd.exe` each time (presumably because the self update never completed). I opened up a MSYS shell, ran `rustup`, and it succeeded. Back in `cmd.exe` it then was fine as well

**Comment by alexcrichton** [comment]
Looks like the same issue as this: https://github.com/brson/multirust/issues/153

**Comment by Diggsey** [comment]
I get this same error when trying to self update.

**Comment by brson** [comment]
I'm wondering if this was the scopeguard problem I already fixed (which you should have after the successful update). I'll be curious to see if you are still getting it tomorrow.

**Comment by brson** [comment]
I have gotten it to error with "error: unable to run updater"

**Comment by Diggsey** [comment]
Maybe we should recommend installing from `rustup.rs` if the self-update fails for whatever reason?

**Comment by Diggsey** [comment]
I still get the same error when running self-update from the latest master - does that contain the scopeguard fix you mentioned?

**Comment by brson** [comment]
@Diggsey yes it does.

**Comment by Diggsey** [comment]
Oh, nevermind it seems to be the downloaded binary which is erroring, so any changes to local don't affect it.

**Comment by brson** [comment]
Hm, if it's the downloaded binary generating the error, then it's probably not the same scopeguard problem since I think that fix is already deployed.

**Comment by brson** [comment]
OK, there are only a few places this error could be generated. It actually could be a PID race I was sure would never happen cause by the way rustup-gc.exe finds its parent process.

**Comment by Diggsey** [comment]
There is potential PID race but it's not the problem, (which I've fixed in #234)
The only way I can see to fix the race is to do a second snapshot after opening the process, and checking that the parent process ID is still the same, and not null.
